eodatasets3/scripts/tostac.py

```diff
@@ -7,70 +7,182 @@
+):
+    """
+    Create STAC document
+
+    :param dataset: Dict of the metadata content
+    :param input_metadata: Path of the Input metadata file
+    :param output_path: Path of the STAC output file
+    :param stac_data: Dict of the static STAC content
+    :param stac_base_url: Base URL for STAC file
+    :param product_base_url: Base URL for ODC product
+    :return: Dict of the STAC content
+    """
+
+    project = partial(
+        pyproj.transform, pyproj.Proj(init=dataset.crs), pyproj.Proj(init="epsg:4326"),
+    )
+    wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
+    item_doc = dict(
+        stac_version=stac_data.get("stac_version", "1.0.0-beta.1"),
+        stac_extensions=stac_data.get("stac_extensions", []),
+        id=dataset.id,
+        type="Feature",
+        bbox=wgs84_geometry.bounds,
+        geometry=wgs84_geometry.__geo_interface__,
+        properties={
+            **{
+                MAPPING_EO3_TO_STAC.get(key, key): convert_value_to_stac_type(key, val)
+                for key, val in dataset.properties.items()
+            },
+            "odc:product": dataset.product.name,
+            "proj:epsg": dataset.crs.lstrip("epsg:"),
+            "proj:shape": dataset.grids["default"].shape,
+            "proj:transform": dataset.grids["default"].transform,
+        },
+        # TODO: Currently assuming no name collisions.
+        assets={
+            **{
+                name: (
+                    {
+                        **stac_data.get("assets", {}).get(name, {}),
+                        "href": urljoin(stac_base_url, m.path),
+                        "proj.shape": dataset.grids[
+                            m.grid if m.grid else "default"
+                        ].shape,
+                        "proj.transform": dataset.grids[
+                            m.grid if m.grid else "default"
+                        ].transform,
+                    }
+                )
+                for name, m in dataset.measurements.items()
+            },
+            **{
+                name: (
+                    {
+                        **stac_data.get("assets", {}).get(name, {}),
+                        "href": urljoin(stac_base_url, m.path),
+                    }
+                )
+                for name, m in dataset.accessories.items()
+            },
+        },
+        links=[
+            {
+                "rel": "self",
+                "type": "application/json",
+                "href": urljoin(stac_base_url, output_path.name),
+            },
+            {
+                "title": "Source Dataset YAML",
+                "rel": "derived_from",
+                "href": urljoin(stac_base_url, input_metadata.name),
+            },
+            {
+                "title": "Open Data Cube Product",
+                "rel": "odc_product",
+                "type": "text/html",
+                "href": urljoin(product_base_url, f"product/{dataset.product.name}"),
```

Comment:
This is now linking to a human-readable html view of the product. I think I'd prefer to keep the "odc_product" rel name for links to the actual machine-readable (yaml/json) product. 

Maybe name it `product_overview` or something else... I'm not sure what the norm is with made-up `rel`s.

Reply:
`odc_product` changed to `product_overview`